Assets/Scripts/Menu Scripts/RadiationButtons.cs

```diff
@@ -29,44 +28,57 @@ void OnClickEvent()
         if (surface_pointcloud)
         {
             Debug.Log("Switching to surface point cloud");
-            WorldProperties.worldObject.GetComponent<ROSDroneConnection>().LampSubscribe_SurfacePointcloud();
+            GameObject sensor = WorldProperties.GetSelectedSensor();
+            LampSensor_ROSSensorConnection lampSensor_ROS = sensor.GetComponent<LampSensor_ROSSensorConnection>();
+            //lampSensor_ROS.Unsubscribe();
         }
 
         if (Level_0)
         {
             Debug.Log("Switching to Level 0");
-            WorldProperties.worldObject.GetComponent<ROSDroneConnection>().LampSubscribe_Colorized_0();
+            GameObject sensor = WorldProperties.GetSelectedSensor();
+            LampSensor_ROSSensorConnection lampSensor_ROS = sensor.GetComponent<LampSensor_ROSSensorConnection>();
+            //lampSensor_ROS.Unsubscribe();
         }
 
         if (Level_1)
         {
             Debug.Log("Switching to Level 1");
-            WorldProperties.worldObject.GetComponent<ROSDroneConnection>().LampSubscribe_Colorized_1();
+            GameObject sensor = WorldProperties.GetSelectedSensor();
+            LampSensor_ROSSensorConnection lampSensor_ROS = sensor.GetComponent<LampSensor_ROSSensorConnection>();
+            //lampSensor_ROS.Unsubscribe();
         }
 
         if (Level_2)
         {
             Debug.Log("Switching to Level 2");
-            WorldProperties.worldObject.GetComponent<ROSDroneConnection>().LampSubscribe_Colorized_2();
+            GameObject sensor = WorldProperties.GetSelectedSensor();
+            LampSensor_ROSSensorConnection lampSensor_ROS = sensor.GetComponent<LampSensor_ROSSensorConnection>();
+            //lampSensor_ROS.Unsubscribe();
         }
 
 
         if (Level_3)
         {
             Debug.Log("Switching to Level 3");
-            WorldProperties.worldObject.GetComponent<ROSDroneConnection>().LampSubscribe_Colorized_3();
+            GameObject sensor = WorldProperties.GetSelectedSensor();
+            LampSensor_ROSSensorConnection lampSensor_ROS = sensor.GetComponent<LampSensor_ROSSensorConnection>();
+            //lampSensor_ROS.Unsubscribe();
         }
 
         if (Level_4)
         {
             Debug.Log("Switching to Level 4");
-            WorldProperties.worldObject.GetComponent<ROSDroneConnection>().LampSubscribe_Colorized_4();
+            GameObject sensor = WorldProperties.GetSelectedSensor();
+            LampSensor_ROSSensorConnection lampSensor_ROS = sensor.GetComponent<LampSensor_ROSSensorConnection>();
+            //lampSensor_ROS.Unsubscribe();
```

Comment:
What is this comment for?

Reply:
And why is it only on levels 0-5 and not 6?

Reply:
Lamp only gives levels 0-5, it was hardcoded to support those. This script will be removed when you implement the dynamic ui.

Reply:
what's the 6th level? @someonejazzy